Zach-zachladwig/Homework-6/4-jquery-ajax-exercise copy/js/main.js

```diff
@@ -0,0 +1,189 @@
+let city;
+let state;
+let country;
+let openWeatherMapTemp;
+let openWeatherMapImg;
+let openWeatherMapDesc;
+let openWeatherMapUrl;
+let weatherUnlockedUrl;
+let weatherUnlockedTemp;
+let weatherUnlockedImg;
+let weatherUnlockedDesc;
+let zip;
+let lat;
+let long;
+
+// define some information we know and won't change
+const units = 'imperial';
+
+// here is where the API keys were but I removed them.
+
+//BONUS 2: Attempting to use form
+$getWeatherButton.on('click', function() {
+  city = $('#city').val();
+  state = $('#state').val();
+  let googleMapsUrl = `https://maps.googleapis.com/maps/api/geocode/json?address=${city},${state}&key=${googleMapsApiKey}`;
+  getLatLong(googleMapsUrl, parseGoogleMaps, setURLs, getWeatherData);
+});
+
+
+function updateUISuccess() {
+  $location1.html(city + ', ' + state);
+  $location2.html(city + ', ' + state);
+  $openWeatherMapTemp.html(openWeatherMapTemp);
+  $openWeatherMapImg.attr("src", openWeatherMapImg);
+  $openWeatherMapDesc.html(openWeatherMapDesc);
+  $weatherUnlockedTemp.html(weatherUnlockedTemp);
+  $weatherUnlockedImg.attr("src", weatherUnlockedImg);
+  $weatherUnlockedDesc.html(weatherUnlockedDesc);
+  $('#city').val('');
+  $('#state').val('');
+}
+
+function updateUIFailure() {
+  $location.html('');
+  $errorMessage.html('That is not a valid city or state. Please try again');
+  $errorMessage.addClass('error');
+}
+
+function parseWeatherUnlocked(data) {
+  let temp = data.temp_f + ' degrees Fahrenheit'
+  let description = data.wx_desc;
+  let icon = data.wx_icon;
+  let img = `set/${data.wx_icon}`;
+  weatherUnlockedTemp = temp;
+  weatherUnlockedImg = img;
+  weatherUnlockedDesc= description;
+};
+
+function parseOpenWeatherMap(data) {
+  let temp = data.main.temp + ' degrees Fahrenheit';
+  let description = data.weather[0].description;
+  let icon = data.weather[0].icon;
+  let img = `http://openweathermap.org/img/w/${data.weather[0].icon}.png`
+  openWeatherMapTemp = temp;
+  openWeatherMapImg = img;
+  openWeatherMapDesc = description;
+};
+
+function parseGoogleMaps(data) {
+  lat = data.results[0].geometry.location.lat;
+  long = data.results[0].geometry.location.lng;
+};
+
+function getLatLong(url, parseData, setURLs, getWeatherData) {
+  return $.get(url, function(data) {})
+  .done(function(data) {
+    parseData(data);
+  })
+  .done(function(data) {
+    setURLs();
+  })
+  .done(function(data) {
+    getWeatherData(weatherUnlockedUrl, parseWeatherUnlocked);
+    getWeatherData(openWeatherMapUrl, parseOpenWeatherMap);
+  })
+  .done(function(data) {
+    updateUISuccess();
+  })
+  .fail(function() {
+    updateUIFailure();
+  })
+};
+
+function setURLs() {
+  openWeatherMapUrl = `http://api.openweathermap.org/data/2.5/weather?lat=${lat}&lon=${long}&APPID=${openWeatherMapApiKey}&units=${units}`;
+  weatherUnlockedUrl = `http://api.weatherunlocked.com/api/current/${lat},${long}?app_id=${weatherUnlockedAppId}&app_key=${weatherUnlockedAppKey}`;
```

Comment:
Where else is `weatherUnlockedAppId` and `weatherUnlockedAppKey` initially declared? I don't see them anywhere else